### ISSUE TITLE
minetest.place_schematic only works once per server session

### ISSUE THREAD

**Issue by MirceaKitsune** [minetest/minetest#8106]
When I import a schematic using minetest.place_schematic, the import process only succeeds once. After that it will not work again until the server is restarted, meaning I must quit to the main menu and start the world again. The error I see in the console per failed attempt is:

`ERROR[Server]: place_schematic: failed to get schematic`

I'm noticing this with my Structures mod, which has a function to export and import mts files. I've ran tests to make sure the file name is always the same, and can confirm that from everything I can tell it shouldn't be anything specific to my mod doing this.

https://github.com/MirceaKitsune/minetest_mods_structures

Minetest is latest Git master from this repository. Compiled and ran on Linux openSUSE Tumbleweed x64. My video card is an AMD GCN 2.0 on amdgpu + Mesa 18.3.1.

**Comment by paramat** [comment]
See #4236 this might be intentional behaviour, not sure.

**Comment by MirceaKitsune** [comment]
Looks like a bug that might caused by that change. My issue isn't that the schematic isn't properly updated, though that too would be a problem: The engine says it can no longer load the file at all after the first attempt, it doesn't even import the first one any more.

**Comment by paramat** [comment]
> Looks like a bug that might caused by that change.

What change?

Please can you describe exactly what your mod does that causes the error, and link to code?
Your structures mod is complex and in at least one way (using place schematic rapidly in 'on generated') not correctly coded, so this might be a modding error.

**Comment by MirceaKitsune** [comment]
I was referring to #4236 regarding the change. I saw it linked to PR #6889 and assumed that might be why my mod is now triggering this problem, but I do not know.

I'm currently using vmanip to place schematics from the mapgen. The error I'm noticing is caused by manually importing structures just once, when the mapgen component isn't active. In my mod the issue can be reproduced by only running the following function twice, be it with the same or different schematics:

https://github.com/MirceaKitsune/minetest_mods_structures/blob/master/structures/structures_io.lua#L92

**Comment by paramat** [comment]
Relevant code:
Error thrown here
https://github.com/minetest/minetest/blob/a122ba0ef4af865d225adbe22a55b8001af04c61/src/script/lua_api/l_mapgen.cpp#L1630

**Comment by paramat** [comment]
Are you using paths in 'filename', for example my own post-mapgen use of 'place schematic':
```
		local path = minetest.get_modpath("projects") .. "/schems/flat.mts" -- spawn flat
		minetest.place_schematic({x=x-6, y=y-1, z=z+1}, path, 90, nil, true)
```

**Comment by MirceaKitsune** [comment]
In my test case, the schematics are in the bin directory, as the mod exports them there and is meant to allow re-importing on the fly. It sounds like that might be a trigger for this issue; Is there a deliberate limitation for importing a file from the path `minetest/bin/myfile.mts`?

I only recently realized that my mod only has this problem when importing structures with the I/O node. When generating cities with the schematics embedded into the mod, it spawns countless of them without ever complaining about missing files. The only difference I can possibly think of is the path.

**Comment by paramat** [comment]
Ah ok good, we're getting somewhere. Not a bug then, i'll make it a suggestion for loading from bin (if possible).

**Comment by MirceaKitsune** [comment]
Wasn't loading from bin always supported? This didn't happen +1 year ago, it only surfaced recently here. But if this behavior is intended then it makes sense why I'm seeing the problem at least. Still odd that it would work once then refuse to import for the rest of that server session though... I figure there's still a bug involved as that doesn't feel quite right.

**Comment by SmallJoker** [comment]
I think I "accidentally" fixed this issue in #11011. If that's not the case, please ask to re-open the issue.
